shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1965,13 +1966,26 @@ - (void)updateEditingStateWithDelta:(flutter::TextEditingDelta)delta {
     @"composingExtent" : @(composingExtent),
   };
 
-  NSDictionary* deltas = @{
-    @"deltas" : @[ deltaToFramework ],
-  };
+  [_pendingDeltas addObject:deltaToFramework];
 
-  [self.textInputDelegate flutterTextInputView:self
-                           updateEditingClient:_textInputClient
-                                     withDelta:deltas];
+  if (_pendingDeltas.count == 1) {
+    CFStringRef runLoopMode = self.customRunLoopMode != nil
+                                  ? (__bridge CFStringRef)self.customRunLoopMode
+                                  : kCFRunLoopCommonModes;
+
+    CFRunLoopPerformBlock(CFRunLoopGetMain(), runLoopMode, ^{
+      if (_pendingDeltas.count > 0) {
```

Comment:
Is it possible for the text input plugin to already be released when the block is invoked?

Reply:
I'm not sure, i'm not too familiar with objc memory management. I would think it is possible since the block is run asynchronously. Is this something we should handle?

Reply:
use `__weak` maybe?